src/Common/src/System/Security/Cryptography/ECDsaSecurityTransforms.cs

```diff
@@ -0,0 +1,544 @@
+        /// </summary>
+        public static new ECDsa Create()
+        {
+            return new ECDsaImplementation.ECDsaSecurityTransforms();
+        }
+
+        /// <summary>
+        /// Creates an instance of the platform specific implementation of the cref="ECDsa" algorithm.
+        /// </summary>
+        /// <param name="curve">
+        /// The <see cref="ECCurve"/> representing the elliptic curve.
+        /// </param>
+        public static ECDsa Create(ECCurve curve)
+        {
+            ECDsa ecdsa = Create();
+            ecdsa.GenerateKey(curve);
+            return ecdsa;
+        }
+
+        /// <summary>
+        /// Creates an instance of the platform specific implementation of the cref="ECDsa" algorithm.
+        /// </summary>
+        /// <param name="parameters">
+        /// The <see cref="ECParameters"/> representing the elliptic curve parameters.
+        /// </param>
+        public static ECDsa Create(ECParameters parameters)
+        {
+            ECDsa ecdsa = Create();
+            ecdsa.ImportParameters(parameters);
+            return ecdsa;
+        }
+#endif
+        internal static partial class ECDsaImplementation
+        {
+            public sealed partial class ECDsaSecurityTransforms : ECDsa
+            {
+                private SecKeyPair _keys;
+
+                public ECDsaSecurityTransforms()
+                {
+                    KeySize = 521;
+                }
+
+                internal ECDsaSecurityTransforms(SafeSecKeyRefHandle publicKey)
+                {
+                    SetKey(SecKeyPair.PublicOnly(publicKey));
+                }
+
+                internal ECDsaSecurityTransforms(SafeSecKeyRefHandle publicKey, SafeSecKeyRefHandle privateKey)
+                {
+                    SetKey(SecKeyPair.PublicPrivatePair(publicKey, privateKey));
+                }
+
+                public override KeySizes[] LegalKeySizes
+                {
+                    get
+                    {
+                        // Return the three sizes that can be explicitly set (for backwards compatibility)
+                        return new[] {
+                            new KeySizes(minSize: 256, maxSize: 384, skipSize: 128),
+                            new KeySizes(minSize: 521, maxSize: 521, skipSize: 0),
+                        };
+                    }
+                }
+
+                public override int KeySize
+                {
+                    get
+                    {
+                        return base.KeySize;
+                    }
+                    set
+                    {
+                        if (KeySize == value)
+                            return;
+
+                        // Set the KeySize before freeing the key so that an invalid value doesn't throw away the key
+                        base.KeySize = value;
+
+                        _keys?.Dispose();
+                        _keys = null;
```

Comment:
Nit: as long as we're doing the null check, we can guard the assignment as well:
```C#
if (_keys != null)
{
    _keys.Dispose();
    _keys = null;
}
```